homeassistant/components/metoffice/data.py

```diff
@@ -65,14 +76,36 @@ async def async_update(self):
     def _update(self):
         """Get the latest data from DataPoint."""
         if self._site is None:
-            _LOGGER.error("No Met Office forecast site held, check logs for problems")
+            _LOGGER.error("No Met Office sites held, check logs for problems")
             return
 
         try:
-            forecast = self._datapoint.get_forecast_for_site(
+            forecast_3hourly = self._datapoint.get_forecast_for_site(
                 self._site.id, MODE_3HOURLY
             )
-            self.now = forecast.now()
+            forecast_daily = self._datapoint.get_forecast_for_site(
```

Comment:
Consider placing the separate api calls into separate methods and then create separate UpdateCoordinators.  This way a failed update in one api call will not prevent data getting unavailable for the other api call.

Reply:
Done